recirq/quantum_chess/dynamic_look_ahead_heuristic_circuit_transformer_test.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+import pytest
+from collections import deque
+
+import cirq
+
+import recirq.quantum_chess.circuit_transformer as ct
+
+a0 = cirq.NamedQubit('a0')
+a1 = cirq.NamedQubit('a1')
+a2 = cirq.NamedQubit('a2')
+a3 = cirq.NamedQubit('a3')
+a4 = cirq.NamedQubit('a4')
+a5 = cirq.NamedQubit('a5')
+a6 = cirq.NamedQubit('a6')
+a7 = cirq.NamedQubit('a7')
+
+logical = {
+    a0: [(a2, 0), (a1, 1), (a3, 6)],
+    a1: [(a0, 1), (a2, 3)],
+    a2: [(a0, 0), (a1, 3), (a3, 5)],
+    a3: [(a2, 5), (a0, 6)],
+}
+
+logical2 = {
+    a0: [(a2, 0), (a1, 1), (a3, 6)],
+    a1: [(a0, 1), (a2, 3), (a7, math.inf)],
+    a2: [(a0, 0), (a1, 3), (a3, 5)],
+    a3: [(a2, 5), (a0, 6)],
+    a4: [(a5, 0), (a6, math.inf)],
+    a5: [(a4, 0)],
+    a6: [(a7, 0), (a4, math.inf)],
+    a7: [(a6, 0), (a1, math.inf)],
+}
```

Comment:
what is "logical" and "logical2"? Expected outputs for some particular test case(s)?
I think it may be better to inline those expectations in the test body so that (1) its clearer to readers what the meaning of these dicts is and (2) readers of the test body won't have to refer back-and-forth to these definitions at a different place in the file

ditto with the 'physical' dict below which I think is used in a similar way for some tests input

Reply:
done, and added more test cases!